### DISCUSSION TITLE
Performance on Laptop vs. Colab

### DISCUSSION THREAD

**Discussion by frankwwu** [pycaret/pycaret#1275] [General]
I run a bunch of tree-based regression model training with PyCaret, and compared running on Colab and a 4-core laptop. It turns out that Colab is not faster, and apparently slower, than the laptop. Colab gives me 2 workers. My discovery is that Colab was faster for Deep Learning with GPU; and had no obviously performance benefits for other Machine Learning algorithms with PyCaret. I am curious to know how PyCaret handles threads for tree models, and what are good practices. Thank you!

**Comment by Yard1** [comment]
PyCaret simply sets the number of available cores for models to the number corresponding in `n_jobs` param passed in `setup`. By default that is `-1`, which would mean "use all available cores".
